File: services/ReadingService1/src/main/java/com/se1853_jv/reading/repository/ReadingWorkflowNoteRepository.java
package com.se1853_jv.reading.repository;

import com.se1853_jv.reading.model.ReadingWorkflowNote;
import com.se1853_jv.reading.model.ReadingWorkflowNoteId;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface ReadingWorkflowNoteRepository extends JpaRepository<ReadingWorkflowNote, ReadingWorkflowNoteId> {

    List<ReadingWorkflowNote> findByIdCollectionIdAndIdPaperIdAndIdUserId(String collectionId, String paperId, String userId);

    Optional<ReadingWorkflowNote> findByIdCollectionIdAndIdPaperIdAndIdUserIdAndIdNoteId(String collectionId, String paperId, String userId, String noteId);
}

File: services/paper-service/src/main/kotlin/com/se1853_jv/repository/ReadingWorkflowNoteRepository.kt
package com.se1853_jv.repository

import com.se1853_jv.model.ReadingWorkflowNote
import com.se1853_jv.model.ReadingWorkflowNoteId
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ReadingWorkflowNoteRepository : JpaRepository<ReadingWorkflowNote, ReadingWorkflowNoteId> {
    
    /**
     * Delete all notes for a specific reading workflow
     */
    fun deleteByRwCollectionIdAndRwPaperIdAndRwUserId(
        collectionId: String,
        paperId: String,
        userId: String
    )
}







